.github/workflows/tests.yml

```diff
@@ -66,6 +66,10 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: run type check
+        run: |
+          mypy conda_smithy tests
```

Comment:
We only check the types at tests?

Reply:
i added type check for the folders 'conda_smithy' and 'tests'. wasn't sure if other files/folders require type check to run as well..